src/core/polyhedral/codegen_llvm.cc

```diff
@@ -294,6 +303,88 @@ class CodeGen_TC : public Halide::Internal::CodeGen_X86 {
 
     value = sym_get(name);
   }
+
+ public:
+  void optimize_module() {
+    Halide::Internal::debug(3) << "Optimizing module\n";
+
+    if (Halide::Internal::debug::debug_level() >= 3) {
+#if LLVM_VERSION_MAJOR >= 5
+      module->print(llvm::dbgs(), nullptr, false, true);
+#else
+      module->dump();
+#endif
+    }
+
+    // We override PassManager::add so that we have an opportunity to
+    // blacklist problematic LLVM passes.
+    class MyFunctionPassManager : public llvm::legacy::FunctionPassManager {
```

Comment:
can you rename it to `LLVMPassManager`

Looking at `MyFunctionPassManager` seems like it is an unimplemented code :)